File: YemenSchoolsV1.Domain/Entities/School.cs
using YemenSchoolsV1.Domain.Commons;
using YemenSchoolsV1.Domain.Enums;

namespace YemenSchoolsV1.Domain.Entities
{
	public class School : GeneralLocalizableEntity
	{
		public Guid Id { get; set; }
		public string NameAr { get; set; }
		public string NameEn { get; set; }
		public string? DescriptionAr { get; set; }
		public string? DescriptionEn { get; set; }
		public string AddressAr { get; set; }
		public string AddressEn { get; set; }
		public string? Logo { get; set; }
		public string? CoverImage { get; set; }
		public string? PostalCode { get; set; } // الرمز البريدي
		public string? MainPhone { get; set; }
		public string Email { get; set; }
		public bool IsActive { get; set; } = true; // حالة المدرسة
		public DateTime? DeactivatedDate { get; set; } // تاريخ إلغاء التفعيل، إذا تم إلغاء التفعيل
		public GenderType GenderType { get; set; } //  فئة الطلاب ,بنين, بنات بنين
		public SchoolType SchoolType { get; set; } // نوع المدرسة (خاصة، حكومية، دولية...)
		public CurriculumType CurriculumType { get; set; } // نوع المنهج
		public SchoolLevel SchoolLevel { get; set; } // مستوى المراحل الدراسية



		//Relation
		public Guid CityId { get; set; } // معرف المدينة
		public Guid RegionId { get; set; } // معرف المنطقة

		public City City { get; set; } = null!;
		public Region Region { get; set; } = null!;

		public ICollection<SchoolNews> SchoolNews { get; set; } = [];
		public ICollection<SchoolPhoto> SchoolPhotos { get; set; } = [];
		public ICollection<SchoolPhone> SchoolPhones { get; set; } = [];
		public ICollection<SchoolRating> SchoolRatings { get; set; } = [];
		public ICollection<Stage> Stages { get; set; } = [];
		public ICollection<Teacher> Teachers { get; set; } = [];
		public ICollection<Subject> Subjects { get; set; } = [];
	}
}

File: YemenSchoolsV1.Clint/src/app/features/schools/models/school.ts
export type { Subject } from "@features/school-dashboard/school-subject/models/subject";

export interface SchoolListItem {
    id: string;
    name: string;
    logo: string | null;
    coverImage: string | null;
    mainPhone: string;
    schoolType: 'Private' | 'Public';
    genderType: 'Boys' | 'Girls' | 'Mixed';
    city: string;
    region: string;
    schoolLevel: string;
    curriculumType: string;
    averageRating: number;


}
export interface SchoolForUpdate {
    id: string;
    nameAr: string;
    nameEn: string;
    addressAr: string;
    addressEn: string;
    postalCode: string;
    mainPhone: string;
    email: string;
    schoolType: number;
    genderType: number;
    curriculumType: number;
    schoolLevel: number;
    cityId: string;
    cityName: string;
    regionId: string;
    regionName: string;
    phoneNumberList: string[];
}
export interface CreateSchoolGradeDto {
    schoolId: string;
    stageGradeIds: string[];
}
export interface StageGradeDto {
    stageGradeId: string;
    stageName?: string;
    gradeName?: string;
    isSelected: boolean;
}
export interface SchoolGradeSubject {
    stageGradeId: string;
    subjectIds: string[];
}
export interface SchoolGradeWithDetailsDto {
    id: string; // هذا هو SchoolGradeId من الباك إند
    stageName?: string;
    gradeName?: string;
    // أضف أي خصائص أخرى إذا كانت موجودة في الـ DTO
}

export interface AssignSubjectsToSchoolGradeDto {
    schoolGradeId: string; // استخدم هذا بدلاً من stageGradeId
    subjectIds: string[];
}
export interface SchoolGradeSubjectsInit {
    stageGradeId: string;
    subjectsIds: string[];
}
export interface SchoolReportData {
    schoolId: string;
    nameAr: string;
    nameEn: string;
    descriptionAr: string | null;
    addressAr: string;
    postalCode: string;
    mainPhone: string;
    email: string;
    schoolType: number; // قد تحتاج إلى تحويلها إلى اسم مفهوم لاحقًا (مثال: 'حكومي', 'خاص')
    schoolLevel: number; // قد تحتاج إلى تحويلها إلى اسم مفهوم
    genderType: number; // قد تحتاج إلى تحويلها إلى اسم مفهوم
    curriculumType: number; // قد تحتاج إلى تحويلها إلى اسم مفهوم
    cityId: string;
    cityNameAr: string;
    regionId: string;
    regionNameAr: string;
    phoneNumbers: any[]; // نوع البيانات حسب ما يأتي من الـ API
    teachersCount: number;
    studentsCount: number;
    gradesCount: number;
    subjectsCount: number;
    sectionsCount: number;
    academicYearsCount: number;
    newsCount: number;
    photosCount: number;
    parentsCount: number;
    ratingsCount: number;
}
export interface SchoolReview {
    id: string;
    schoolId: string;
    userId: string;
    userName: string;
    userImage: string;
    rating: number;
    comment?: string;
    createdAt: string;
}export interface SchoolPhoto {
    id: string;           // GUID
    schoolId: string;     // GUID للمدرسة
    photoUrl: string;     // رابط الصورة

}
